### PR TITLE
Implement test instance of jenkins for local iteration on global jenk…

### PR DESCRIPTION
**Pull request by gileshinchcliff**
…ins pipeline.

### CODE REVIEW

File: jenkins/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+RUN apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D && \
+    sh -c "echo deb https://apt.dockerproject.org/repo debian-jessie main > /etc/apt/sources.list.d/docker.list" && \
+    apt-get update && \
+    apt-get install -y --force-yes docker-engine=1.11.0-0~jessie && \
+    pip install --upgrade cffi==1.7.0 botocore awscli
+
+RUN echo 2.0 > /usr/share/jenkins/ref/jenkins.install.UpgradeWizard.state
+
+#COPY credentials.xml /app/credentials.xml
```
Comment (ajohnstone) [inline jenkins/Dockerfile]: Remove comments.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+
+Also we assume docker is installed
+
+### Environment Variables
+
+- `DEBUG_BRANCH` the branch you want to test against, this will manage update jenkins global libraries plugin to map against your branch.
+
+- `SSH_USERNAME` The username you use for connecting to github.
+
+- `PERSONAL_ACCESS_TOKEN` The personal access token retrieved from github, this need to have full access to the repo permission to enable organisations. see https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/
```
Comment (ajohnstone) [inline README.md]: What permissions are required?

File: jenkins/run.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+if [ -z "${SSH_USERNAME}" ]; then
+   echo "Please set SSH_USERNAME to your github username"
+   exit 1
+fi
+if [ -z "${PERSONAL_ACCESS_TOKEN}" ]; then
+   echo "Please set your github PERSONAL_ACCESS_TOKEN"
+   exit 1
+fi
+if [ -z "${DEBUG_BRANCH}" ]; then
```
Comment (gileshinchcliff) [inline jenkins/run.sh]: this is fixed...

File: README.md (hunk 2 above)
Comment (gileshinchcliff) [inline README.md]: the repo permission as stated.

**Pull request merged**